app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -258,9 +284,16 @@ export default function createRPCMethodTrackingMiddleware({
         properties: eventProperties,
       });
 
-      rateLimitTimeouts[method] = setTimeout(() => {
-        delete rateLimitTimeouts[method];
-      }, SECOND * rateLimitSeconds);
+      if (rateLimitType === RATE_LIMIT_TYPES.TIMEOUT) {
+        rateLimitTimeoutsByMethod[method] = setTimeout(() => {
+          delete rateLimitTimeoutsByMethod[method];
+        }, rateLimitTimeout);
+      }
+
+      globalRateLimitCount += 1;
+      setTimeout(() => {
```

Comment:
This new global timeout looks good to me 👍 

In the worst case, 10 events in a span of 5 minutes should have a roughly similar volume to that of the old strategy of rate limiting individual methods.